### DISCUSSION TITLE
SDXL Lora training on runpod

### DISCUSSION THREAD

**Discussion by meanin2** [TheLastBen/fast-stable-diffusion#2452] [Q&A] [answered]
Hi, first off shout out to @TheLastBen I love your work!
Thank you so much for making training accessible for everyone!

A few questions about sdxl lora training in runpod:

For image size can I use different sized images and set crop to false? (Above 1024x1024 but all different sizes and ratios)

Does image count matter? (Or is it the same 10 you always recommended for the collab notebook?)

Where can I upload my created lora to hugging face? (I dont see a save or upload cell anywhere in the sdxl training notebook)

Is there minimum requirement for training a sdxl lora? (ram, vram, etc)

Thank you so much

**Comment by TheLastBen** [comment, marked as answer]
Hi and thanks,
yes you can use any size you want, make sure it's 1:1

10-20 images are enough to inject the concept into the model.

since LoRA files are not that large, I removed the hf upload cell.

3090 is good for LoRA training

**Reply by meanin2** [reply]
Amazing, thank you so much. 
For clarity, 10-20 should be enough but i can use as many as i want its just unnecessary, 
or 10 -20 is the right amount for the lr and default settings in the notebook, any more will mess with it?

**Reply by TheLastBen** [reply]
you can try more, but you'll have to experiment with different settings in that case

**Reply by meanin2** [reply]
thanks, last question, 
In the loras I trained so far, the file size ended up being around 1.3gb shouldn't they be smaller?

**Reply by TheLastBen** [reply]
reduce the dim to 64 or 32, but the quality will decrease a bit

**Reply by THX1139b** [reply]
So the default epoch values are unchanged for either 10 or 20 images? Or should I double both numbers for 20 images (240 + 80 text) ?Thanks!

**Reply by TheLastBen** [reply]
if the number of images is higher, reduce the number of epochs to keep the total number of steps in the 1500-3000 range

**Reply by THX1139b** [reply]
Aha, I have clearly thought about epochs the wrong way. Thank you!
